### ISSUE TITLE
Remove ads from Plugin Manager

### ISSUE THREAD

**Issue by xchannels** [bruderstein/nppPluginManager#80]
Can you remove ads from Plugin Manager so the developers of Notepad++ can bring back Plugin Manager again?

**Comment by bruderstein** [comment]
Can you provide a server and bandwidth to host the plugin list on? That's what the "ad" (it's just a logo) provides.

See the comment here for more info
https://notepad-plus-plus.org/community/topic/14315/notepad-7-5-release/34

**Comment by xchannels** [comment]
I'm quite unsure what you mean but i'm disappointed with Notepad's removal of Plugin Manager. That's why i asked. So much Google-feature-remover-copycats now a days

**Comment by sm4rk0** [comment]
@bruderstein, reading that thread, to me it seems like Don Ho is ignoring everything nppPluginManager-related.

Re: hosting
How about clikcing a random link [here](https://download.savannah.gnu.org/mirmon/allgnu/) and contacting the stuff? Both npp and the plugin manager are free and open source.

**Comment by AyrA** [comment]
> Can you provide a server and bandwidth to host the plugin list on? That's what the "ad" (it's just a logo) provides.

What's the bandwidth requirement? It doesn't sounds like too big of a deal. The Zip files (x64 + x86) are 70 kb in size together. With my bandwidth I could save them at least 500 times per second.

**Comment by bruderstein** [comment]
Bandwidth requirements are in the link above (https://notepad-plus-plus.org/community/topic/14315/notepad-7-5-release/34), but basically cloudflare is sitting in front of it, so that's taking most of the traffic - it's about 150 to 250Gb transfer a month (the files are really small, but there are a _lot_ of users).  What actually lands on the server is about 1-2Gb per month. But, if cloudflare ever became not-free (which would be perfectly reasonable - they're providing quite a bit of infra for free currently), we'd need to be able to support that 150-250Gb per month. That's what the current hosting arrangement provides.

**Comment by AyrA** [comment]
In that case it's time you get a hosting agreement without limited bandwidth. If you want, I have a server sitting here that is able to push a constant 50 mbit/s. This results in a possible traffic of 16 TB a month.

**Comment by bruderstein** [comment]
That's effectively what we've got - the cost is that link on the plugin window that caused all the fuss.

I'm not quite sure what you're saying - are you offering the project free hosting (in which case, that would be totally awesome!), or are you suggesting we look for a deal like that? I don't want to pay $10-20 per month out of my own pocket to host it - that's why the hosting deal came about.

**Comment by AyrA** [comment]
I can host it, no problem.

**Comment by bruderstein** [comment]
Awesome! Can you send me a mail so we can work out details?  davegb at pobox dot com.

**Comment by AyrA** [comment]
I sent you a message

**Comment by vlude** [comment]
250gb per month is a really small requirement nowadays, so that should be easy to arrange. I could also provide it, if it's still necessary

**Comment by GitMensch** [comment]
Are there any updates on this?

**Comment by AyrA** [comment]
@GitMensch Yes, the plugin list will be hosted elsewhere in the future, but the author is busy at the moment so it might take a month or two until the new system is fully set up.

**Comment by psylenced** [comment]
A lot of open source use github for hosting updated list / config files.

You commit the change - the list is automatically updated.

No hosting costs.

**Comment by AyrA** [comment]
@psylenced But the data is often in text format, for example adblocker lists. In this case the data is (apart from the index) entirely of binary nature. Github is probably not the right platform for that and they might even forbid doing this.

**Comment by GitMensch** [comment]
@AyrA GitHub should be fine as long as the shipped files are binaries compiled from source that meets the requirements (no closed-source).

See https://github.com/search?q=.dll&type=Commits and you see bunch of dll commits...

Actually it would be very good to have all the binaries in the repo (and git handles binary data quite well) as this allows to go back to any version number uploaded ever.
Thinking about this more: There should be a feature request to be able to select older versions (it happened more than once and is understandable that funky new version X provided bug fixes for b1, b2, b3 and enhancements, therefore you click on "download", just to recognize you like these enhancements but there is a new bug b4 which is much more severe for you - and the plugin manager currently has no way of getting the old version back).

**Comment by AyrA** [comment]
Just because people commit dll files doesn't makes it the right thing to do.

Iirc github was once unhappy with certain adblock lists being hosted there and being requested daily by millions of users around the globe. Not sure if ti is an issue anymore.

If you want to be able to rollback your plugins backup them.

**Comment by GitMensch** [comment]
There is no explicit "don't do it" in [Github's Terms of Service][1] and binaries in general are permitted, too.

The only thing that *could* be an issue is [C7][2]

># Excessive Bandwidth Use
> If we determine your bandwidth usage to be significantly excessive in relation to other GitHub customers, we reserve the right to suspend your account or throttle your file hosting until you can reduce your bandwidth consumption.

In general: If this would be a possibility for nppPluginManager we can ask beforehand to check if it is acceptable for GitHub or not.
But as there were other solutions mentioned we currently don't have a reason to ask.

My FR about "choosing the version of a plugin" stand, I'll add a FR ticket for this asap. 

[1]:https://help.github.com/articles/github-terms-of-service
[2]:https://help.github.com/articles/github-terms-of-service/#7-excessive-bandwidth-use

**Comment by bruderstein** [comment]
We're almost sorted with the server from @AyrA now (will probably switch in a few days and release a new version without the logo & link). Whether Don decides to carry on with the internal version and some form of git hosting rather than contributing here will be key to whether this project is worth maintaining.

I've mentioned before about GitHub - I asked them about this specifically and they quoted me $80/month.

**Comment by GitMensch** [comment]
Thank you for the update (I didn't knew about your GitHub message).

**Comment by DeepMac** [comment]
@bruderstein I'd like to volunteer to provide an additional mirror for the plugin list, in addition to @AyrA 
I've got an unlimited bandwidth and disk space account with my provider. As far as I can tell it'd not be any problem at all. You can contact me if you're interested.

**Comment by alexhass** [comment]
Are the files on these untrusted servers protected from manupulation so that manipulated things are not getting distributed automatically.

**Comment by CTimmerman** [comment]
Didn't BitTorrent fix lack of bandwidth years ago? It works for LibreOffice and lots of other free software.

**Comment by AyrA** [comment]
@CTimmerman Notepad++ is a development tool which means a large number of users run it inside a corporate network. BitTorrent can create problems in these situations. It's blocked in many corporate networks and your sysadmin might get mad at you for running a BitTorrent application.

The Solution we are getting together doesn't has this problem anymore. CloudFlare will sit in front of the server which takes most of the bandwidth requirements away, but the author still wants a server that is capable of handling everything on its own in case Cloudflare decides to bail out.

**Comment by GitMensch** [comment]
Please try to provide a fallback server (with an "approximation of zero bandwidth in normal months"), too.
The recent SF outtakes are a good sample why it is bad if you don't have any fallback for downloads with another hosting provider.

**Comment by AyrA** [comment]
@GitMensch That's essentially what the cloudflare cache does.

**Comment by Nazgul07** [comment]
Why not just put the list in a github repo (json file or such) and have the manager pull it from "released" versions?

**Comment by tberneman** [comment]
With the new 64bit Plugin Manager will there be any issues installing plugins since they weren't originally written for 64bit?

Also, where's the "TextFX" plugin?

**Comment by SuperSandro2000** [comment]
Did anyone ever think about using GitHub releases and a boost the urls in a file on GitHub?

**Comment by GitMensch** [comment]
@SuperSandro2000 See notes in this issue you've just responded to read about people thinking about this and the reasons that it wasn't accepted.

**Comment by GitMensch** [comment]
@tberneman 

> With the new 64bit Plugin Manager will there be any issues installing plugins since they weren't originally written for 64bit?

A plugin that isn't *compiled* for 64bit can simply not run at all - and therefore the plugin manager (which doesn't have the plugin for 64 bit) won't show them.

> Also, where's the "TextFX" plugin?

... if you use the 64bit version the plugin may be not compiled, if you use the 32bit version it isn't available, otherwise it would be shown (my 32bit version shows TextFX Characters" (may be the referenced plugin, or not...)

**Comment by barrychapman** [comment]
is this thing moving forward? are the 64bit binaries available anywhere?

**Comment by CTimmerman** [comment]
64 bit Plugin Manager is now available #3264

**Comment by divinity76** [comment]
$80/month for 150-250Gb/month is ridiculously expensive.  i pay $20/month for a server that pushes ~25-30TB/month...

**Comment by GitMensch** [comment]
@divinity76 Is this an offer to host the plugins for the next years if the server from @AyrA isn't available anymore (and if it still is then offering to provide a backup-server that the Plugin Manager could use if the other server isn't available)?
And actually GitHub may have a different view on this now after the take-over from MS (GitHub can quote whatever they like; "ridiculously expensive" and "being very 'important'/well known" tend to get together quite often).

@bruderstein I don't want to push you but what happened to the following quote from Feb-21?

> We're almost sorted with the server from @AyrA now (will probably switch in a few days and release a new version without the logo & link).

**Comment by divinity76** [comment]
>  Is this an offer to host the plugins for the next years if the server from @AyrA isn't available anymore (and if it still is then offering to provide a backup-server that the Plugin Manager could use if the other server isn't available)?

depends on how much disk space is required (<50GB shouldn't be a problem), but yeah i could probably do that, on a few technical conditions. 
(you won't get a dedicated ip, and you won't get a dedicated port 80/443, those 2 ports will go through a nginx reverse-proxy to the nodejs server, and you won't get true root access, albeit you can get root access to a VPS if needed, albeit i'd prefer if you could just make due with a user account on the (debian linux) system and let me know what root-actions needs to be performed~ )

i just didn't come with the offer up front because it seems several others already beat me to it~
